src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs

```diff
@@ -78,7 +82,8 @@ private static (IReadOnlyDictionary<FieldDeclaration, InputModelProperty> Fields
 
             foreach (var inputModelProperty in inputModel.Properties)
             {
-                var fieldModifiers = inputModelProperty.IsReadOnly || inputModelProperty.Type is InputDictionaryType or InputListType
+                var fieldModifiers = inputModelProperty.IsReadOnly || inputModelProperty.Type is InputDictionaryType or InputListType ||
+                    (!inputModel.Usage.HasFlag(InputModelTypeUsage.Output) && inputModelProperty.IsRequired)
```

Comment:
Will you exclude InputModelTypeUsage.None?

Reply:
I actually have the same question. In existing codes, there are patterns like `!_usage.Flag(SchemaTypeUsage.Input)`:https://github.com/Azure/autorest.csharp/blob/cf3f4bef2e1fdb860ffb282a45b2aa491d88a43e/src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs#L476-L477

I think that way is better if in the future the flags will extend. Right now there are only 4 flags, but if you check M4 schema usage, there are 8 flags: https://github.com/Azure/autorest.csharp/blob/cf3f4bef2e1fdb860ffb282a45b2aa491d88a43e/src/AutoRest.CSharp/Common/Input/SchemaUsageProvider.cs#L170-L180

Reply:
`SchemaTypeUsage` breaks SRP: `Input`/`Output`/`RoundTrip` has nothing to do with `FlattenedParameters`, which is not even a usage. So no, `InputModelTypeUsage` will not become bigger, and support for model customization will be done differently.